packages/next-sample/pages/profile-ssr.tsx

```diff
@@ -0,0 +1,64 @@
+import { LogtoUser } from '@logto/next';
+import { useMemo } from 'react';
+
+import { logtoClient } from '../libraries/logto';
+
+type Props = {
+  user: LogtoUser;
+};
+
+const ProfileSsr = ({ user }: Props) => {
+  const userInfo = useMemo(() => {
+    if (!user.isAuthenticated || !user.claims) {
+      return null;
+    }
+
+    return (
+      <div>
+        <h2>User info:</h2>
+        <table>
+          <thead>
+            <tr>
+              <th>Name</th>
+              <th>Value</th>
+            </tr>
+          </thead>
+          <tbody>
+            {Object.entries(user.claims).map(([key, value]) => (
+              <tr key={key}>
+                <td>{key}</td>
+                <td>{value}</td>
+              </tr>
+            ))}
+          </tbody>
+        </table>
+      </div>
+    );
+  }, [user]);
+
+  return (
+    <div>
+      <header>
+        <h1>Hello Logto.</h1>
+      </header>
+      {userInfo}
+    </div>
+  );
+};
+
+export default ProfileSsr;
+
+export const getServerSideProps = logtoClient.withLogtoSsr(async function ({ req, res }) {
+  const { user } = req;
+
+  if (!user.isAuthenticated) {
+    res.setHeader('location', '/api/login');
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    res.statusCode = 302;
+    res.end();
+  }
+
+  return {
+    props: { user },
+  };
+});
```

Comment:
1. Maybe we should put this on top of the file?
2. Seems we don't need to `export`

Reply:
export this function as a requirement of Next.js, https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props

Reply:
Also, put this after "default export(component)" is common.